src/main/java/net/neoforged/neoforge/network/PacketDistributor.java

```diff
@@ -167,18 +157,26 @@ public void send(Packet<?> packet) {
             packetConsumer.accept(packet);
         }
 
-        public PlayNetworkDirection getDirection() {
-            return distributor.direction;
+        public void send(CustomPacketPayload payload) {
```

Comment:
Is it possible to make this vararg and if there is one element just send the packet and otherwise behave as if it was `IServerGamePacketListenerExtension#sendBundled`? That way the packet distributors don't have to be reimplemented for sending a bundled packet.

Reply:
Sadly no, due to the way PacketDistributor is currently structured this is not easy. It is simply only intended to send one payload/packet at a time.

Reply:
Statement was mistaken :dagger: 

I implemented it now as follows:
- If client bound and with more then 1 packet, create bundle.
- If client bound and one packet, send one packet
- If server bound send packets individually without bundle.